test/unit/support.d

```diff
@@ -91,3 +91,238 @@ class NoopDiagnosticReporter : DiagnosticReporter
     override void deprecation(const ref Loc loc, const(char)* format, va_list) {}
     override void deprecationSupplemental(const ref Loc loc, const(char)* format, va_list) {}
 }
+
```

Comment:
There seems to be 150 lines of code here doing what? And what does it have to do with opEquals?

Reply:
It collects all the diagnostic messages reported by the lexer. The `toString` methods converts them to a string which is used in the assert message if the lexer encounters an error.

Reply:
Added some documentation.